### PR TITLE
add documents for Learner.get_preds

### LINKED ISSUE #3676: Docment `learn.get_preds`
This is a nice first issue:

Add types/docments to the `Learner.get_preds` function. This function is essential to any `fastai` user and has almost no documentation. Add types and text to the variables as we have in many places now.

### PR DESCRIPTION
**Pull request by kurianbenoy**
Fixes #3676

### CODE REVIEW

**Comment by jph00** [comment]
Thanks for the PR. Have a closer look at the source code to see how `dl` is used -- your current description isn't correct.

**Comment by jph00** [comment]
@kurianbenoy would you like to fix the conflicts and remove "draft" attribute when you feel this is ready to merge?

**Comment by kurianbenoy** [comment]
I would like to fix the conflicts and get with a ready to merge version pretty soon. I am hoping to complete it by this week sometime.

File: fastai/learner.py [hunk 1]
```diff
@@ -271,8 +271,19 @@ def validate(self, ds_idx=1, dl=None, cbs=None):
...
+                  dl: TfmdDL | None=None, # Dataloader to be passed with ds_idx
+                  with_input: bool=False, # Whether to return inputs
+                  with_decoded: bool=False, # Whether to decode based on loss function passed
+                  with_loss : bool=False, # Whether to return losses
+                  act: Any=None, # Option to pass Activation function to predict function
+                  inner : bool=False, # Tells that it's used internally used anywhere like in another training loop
+                  reorder : bool=True, # To order the tensors appropriately
+                  cbs: list | None =None, # Option to pass `Callbacks` to predict function
+                  **kwargs
+                 )-> tuple[tensor, tensor| None]:
```
Comment (jph00) [inline fastai/learner.py:285]:
```suggestion
                 )-> tuple:
```

We don't use generic types in fast.ai libs.

**Review by jph00 — CHANGES_REQUESTED** [review]

File: fastai/learner.py (hunk 1 above)
Comment (kurianbenoy) [reply fastai/learner.py:285]: Noted. Thanks

**Comment by jph00** [comment]
Don't forget to update the notebook @kurianbenoy ! :D

**Comment by kurianbenoy** [comment]
Looks like I messed up with my latest changes to update description.

**Comment by jph00** [comment]
If it's easier to close this PR and open a new clean one with your changes, feel free to do that @kurianbenoy . Or I could paste them manually into the notebook at my end if you prefer.

**Comment by kurianbenoy** [comment]
Thanks for your suggestion Jeremy. Following your suggestion, I have opened a new PR with my changes in #3798 as it felt easier to do that way.

Closing #3718

**Pull request closed without merging**